### PR TITLE
Moved alerts to api-demos, find tests

### PR DESCRIPTION
**Pull request by scottdixon**

### CODE REVIEW

**Comment by scottdixon** [comment]
@moizjv

**Comment by moizjv** [comment]
awesome coverage!!!! Can you look at this https://team-appium.ci.cloudbees.com/job/Appium-android-driver/18/console?

**Comment by moizjv** [comment]
All green! CI works :)

File: test/functional/api-demos/find/find-basic-e2e-specs.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  });
+  it('should fail on empty locator', async () => {
+    await driver.findElOrEls('class name', '', true).should.be.rejectedWith(/selector/);
+  });
+  it('should find a single element by string id @skip-android-all', async () => {
+    let el = await driver.findElOrEls('id', 'activity_sample_code', false);
+    await driver.getText(el.ELEMENT).should.eventually.equal('API Demos');
+  });
+  it('should find a single element by resource-id', async () => {
+   await driver.findElOrEls('id', `android:id/${singleResourceId}`, false)
```
Comment (moizjv) [inline test/functional/api-demos/find/find-basic-e2e-specs.js]: indentation

File: test/functional/api-demos/find/find-basic-e2e-specs.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  it('should find a single element by resource-id with implicit package', async () => {
+   await driver.findElOrEls('id', singleResourceId, false)
+          .should.eventually.exist;
+  });
+  it('should find a single element by resource-id with implicit package', async () => {
+   await driver.findElOrEls('id', 'text1', true)
+          .should.eventually.have.length.at.least(10);
+  });
+  it('should find multiple elements by resource-id with implicit package even when theres just one', async () => {
+   await driver.findElOrEls('id', singleResourceId, true)
```
Comment (moizjv) [inline test/functional/api-demos/find/find-basic-e2e-specs.js]: same here and above

File: test/functional/driver-e2e-specs.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ let defaultCaps = {
 };
 
 describe('createSession', function () {
+  this.timeout(20000);
```
Comment (moizjv) [inline test/functional/driver-e2e-specs.js]: do we need 20 sec?

File: test/functional/api-demos/find/by-uiautomator-e2e-specs.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  });
+  after(async () => {
+    await driver.deleteSession();
+  });
+  it('should find elements with a boolean argument', async () => {
+    await driver.findElOrEls('-android uiautomator', 'new UiSelector().clickable(true)', true)
+            .should.eventually.have.length.at.least(10);
+  });
+  it('should find elements within the context of another element', async () => {
+    let els = await driver.findElOrEls('-android uiautomator', 'new UiSelector().className("android.widget.TextView")', true);
```
Comment (moizjv) [inline test/functional/api-demos/find/by-uiautomator-e2e-specs.js]: may be more than 80 characters

File: test/functional/api-demos/find/by-uiautomator-e2e-specs.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  it('should ignore trailing semicolons', async () => {
+    await driver.findElOrEls('-android uiautomator', 'new UiSelector().clickable(true);', true)
+            .should.eventually.have.length.at.least(10);
+  });
+  it('should find an element with an int argument', async () => {
+    let el = await driver.findElOrEls('-android uiautomator', 'new UiSelector().index(0)', false);
+    await driver.getName(el.ELEMENT).should.eventually.equal('android.widget.FrameLayout');
+  });
+  it('should find an element with a string argument', async () => {
+    await driver.findElOrEls('-android uiautomator', 'new UiSelector().description("Animation")', false).should.eventually.exist;
```
Comment (moizjv) [inline test/functional/api-demos/find/by-uiautomator-e2e-specs.js]: here too > 80 characters

**Comment by scottdixon** [comment]
@moiz @imurchie

**Pull request merged**